integrations/firebase_crashlytics_demo_project/README.md

```diff
@@ -1,40 +1,94 @@
+## Getting the dSYM file
+
+As soon as your build finishes successfully, debug symbols are generated. However, if you want them to be displayed in the Codemagic UI on the build page, then the following path needs to be configured in `codemagic.yaml` under the artifacts section:
+
+```yaml
+  artifacts:
+    - $HOME/Library/Developer/Xcode/DerivedData/**/Build/**/*.dSYM
+```
+
+## Uploading to Firebase Crashlytics
+
+### Flutter
+
+In order to upload the dSYM files to Firebase Crashlytics, add the following script to your `codemagic.yaml` configuration file or to your post-publish script in the Flutter workflow editor: 
+
+```yaml
+publishing:
+  scripts:
+    - name: Upload debug symbols to Firebase Crashlytics
+      script: | 
+        echo "Find build artifacts"
+        dsymPath=$(find $CM_BUILD_DIR/build/ios/archive/Runner.xcarchive -name "*.dSYM" | head -1)
+        if [[ -z ${dsymPath} ]]
+        then
+          echo "No debug symbols were found, skip publishing to Firebase Crashlytics"
+        else
+          echo "Publishing debug symbols from $dsymPath to Firebase Crashlytics"
+          ls -d -- ios/Pods/*
+          $CM_BUILD_DIR/ios/Pods/FirebaseCrashlytics/upload-symbols \
+            -gsp ios/Runner/GoogleService-Info.plist -p ios $dsymPath
+        fi
 ```
- - $HOME/Library/Developer/Xcode/DerivedData/**/Build/**/*.dSYM
+
+### Others
+
+The above-mentioned **dsymPath** is Flutter specific and it could change depending on what platform the app is built. E.g., for React Native or Native iOS applications you might use:
+
+```yaml
+dsymPath=$(find $CM_BUILD_DIR/build/ios/xcarchive/*.xcarchive -name "*.dSYM" | head -1)
 ```
-More information about **codemagic.yaml** can be found [here](https://docs.codemagic.io/yaml/yaml-getting-started/)
-
-With the help of the following bash script that needs to be present under a post-publishing script (a script that needs to be run after building your artifacts binaries), **dSYM** files are found and uploaded to Firebase Crashlytics:
-
-  ```bash
-  echo "Find build artifacts"
-  dsymPath=$(find $CM_BUILD_DIR/build/ios/archive/Runner.xcarchive -name "*.dSYM" | head -1)
-  if [[ -z ${dsymPath} ]]
-  then
-    echo "No debug symbols were found, skip publishing to Firebase Crashlytics"
-  else
-    echo "Publishing debug symbols from $dsymPath to Firebase Crashlytics"
-    ls -d -- ios/Pods/*
-    $CM_BUILD_DIR/ios/Pods/FirebaseCrashlytics/upload-symbols -gsp ios/Runner/GoogleService-Info.plist -p ios $dsymPath
-  fi
-  ```
+
+If necessary, you can use remote access to the build machine to find the correct path. More information can be found [here](https://docs.codemagic.io/troubleshooting/accessing-builder-machine-via-ssh).
+
+
+## iOS apps using SwiftPackageManager (SPM)
+
+For Native iOS apps, in the case of using SwiftPackageManager (SPM) instead of CocoaPods, the following script needs to be added in a post-publishing script:
```

Comment:
```suggestion
For native iOS apps, in the case of using SwiftPackageManager (SPM) instead of CocoaPods, the following script needs to be added in a post-publishing script:
```